app/services/transaction.service.js

```diff
@@ -90,28 +90,36 @@ const refund = function refundTransaction (gatewayAccountId, chargeId, amount, r
   logContext[keys.CORRELATION_ID] = correlationId
   logger.log('info', 'Submitting a refund for a charge', logContext)
 
-  const params = { gatewayAccountId, chargeId, payload, correlationId }
+  const payload = {
+    amount: amount,
+    refund_amount_available: refundAmountAvailable,
+    user_external_id: userExternalId,
+    user_email: userEmail
+  }
 
-  return new Promise(function (resolve, reject) {
-    connector.postChargeRefund(params, function () {
-      resolve()
-    }).on('connectorError', (err, response, body) => {
-      err = 'REFUND_FAILED'
-      if (response && response.statusCode === 400) {
-        if (body.reason) {
-          err = body.reason
-        }
+  try {
+    await connector.postChargeRefund(gatewayAccountId, chargeId, payload, correlationId)
```

Comment:
🎉  so much clearer, replacing use of the `oldBaseClient` makes upstream code 10x easier to read.